builtin/common/tests/vector_spec.lua

```diff
@@ -38,9 +63,45 @@ describe("vector", function()
 			local a = { x = 2, y = 4, z = -10 }
 			assertE(a, a)
 			assertNE({x = -1, y = 0, z = 1}, a)
+
+			assert.equal(vector.new(1, 2, 3), vector.new(1, 2, 3))
 	end)
 
 	it("add()", function()
-		assert.same({ x = 2, y = 4, z = 6 }, vector.add(vector.new(1, 2, 3), { x = 1, y = 2, z = 3 }))
+		assert.equal(vector.new(2, 4, 6), vector.add(vector.new(1, 2, 3), { x = 1, y = 2, z = 3 }))
+	end)
+
+	it("length()", function()
+		assert.equal(0, vector.length(vector.new()))
+		assert.equal(23, vector.length(vector.new(0, 0, -23)))
+	end)
+
+	it("operators", function()
+		local a = vector.new(1, 2, 3)
+		local b = vector.new(1123, 232, 3234)
+
+		assert.equal(a, vector.new(a))
+
+		assert.equal(vector.add(a, b), a + b)
+		assert.equal(vector.add(a, b), a:add(b))
+		assert.equal(a + b, b + a)
+		assert.equal(a + 1, 1 + a)
+
+		assert.equal(vector.subtract(a, b), a - b)
```

Comment:
what about `a:subtract(...)`?

Reply:
The tests aren't meant to be covering as they weren't before either.
Testing one function with `:` should suffice.

Reply:
But it would be nice improve coverage.

Reply:
Done.
But I won't add tests for multiply and co., it would blow up this PR even more.